prottp.go

```diff
@@ -37,10 +40,22 @@ func Wrap(service Service) http.Handler {
 	return mux
 }
 
+var marshaler = jsonpb.Marshaler{
+	EnumsAsInts:  false,
+	EmitDefaults: false,
+	Indent:       "\t",
+	OrigName:     true,
+}
+
 func wrapMethod(service interface{}, m grpc.MethodDesc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		isJson := strings.Index(r.Header.Get("Content-Type"), "application/json") >= 0
 		//func _SearchService_Search_Handler(
 		dec := func(i interface{}) error {
+			if isJson {
+				return jsonpb.Unmarshal(r.Body, i.(proto.Message))
```

Comment:
Suggest using [`Unmarshaler{}.Unmarshal`](https://godoc.org/github.com/golang/protobuf/jsonpb#Unmarshaler) as that will set `AllowUnknownFields` to false. It's better to be strict.

The package function is documented as: 

> This function is lenient and will decode any options permutations of the related Marshaler.

I *suspect*, but am not sure, that this means `AllowUnknownFields: true`.